File: lib/qt4/AbilityListComboBox.cpp
/*
 * Copyright (c) 2013-2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

#include <string>
#include <vector>

#include <pkmn/lists.hpp>
#include <pkmn/qt4/AbilityListComboBox.hpp>

namespace pkmn
{
    namespace qt4
    {
        AbilityListComboBox::AbilityListComboBox(const uint16_t gen, QWidget* parent):
            QComboBox(parent)
        {
            std::vector<pkstring> abilities_vec;
            get_ability_list(abilities_vec, gen);

            for(uint16_t i = 0; i < abilities_vec.size(); i++)
                addItem(QString::fromUtf16(abilities_vec[i]), QVariant(i));
        }
    }
} /* namespace pkmn */

File: lib/swig/java/ComboBoxes/AbilityListComboBox.java
/*
 * Copyright (c) 2013-2015 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.*;

/**
 * A ComboBox that lists all abilities available in the given generation.
 *
 * @param gen highest generation whose abilities to include
 */
public class AbilityListComboBox extends JComboBox<String>
{
    public AbilityListComboBox(int gen)
    {
        super();

        StringVector abilityVec = new StringVector();
        Lists.getAbilityList(abilityVec, gen);
        for(int i = 0; i < abilityVec.size(); i++)
        {
            addItem(abilityVec.get(i));
        }

        setEditable(false);
    }
}
